benchmarks/arrow_parquet-arrow-fuzz/Dockerfile

```diff
@@ -28,6 +27,9 @@ RUN apt-get update -y -q && \
         ninja-build \
         python3
 
-RUN git clone --depth=1 https://github.com/apache/arrow.git $SRC/arrow
+RUN git clone \
+        --depth=1 \
+        --branch apache-arrow-10.0.0 \
```

Comment:
Sadly I don't think thsi approach youve taken really works with the bug based benchmarks because these benchmarks are only useful when a particular version is checked out. Otherwise they wont have the bugs we expect.

Reply:
After our discussion today, I was thinking about the following, but not sure if it would be a good idea:
1. We build **all** benchmarks **only** with the `Dockerfile` for coverage experiments.
2. If the benchmark specifies a commit in its config file, we first build it with the `Dockerfile` and then checkout to the commit, as use it for bug experiments.

In that way, we can use all bug benchmarks for coverage (we have already supported them anyway), and the commit in the config file is no longer a redundant design.

Reply:
Ah sorry, 1 means not using the commit hash in `project.yaml`. It will still need `build.sh`.

The intention is to make use of current bug benchmark programs (especially the ones that are upgradable) for code-coverage evaluation, and the commit in `project.yaml` for bug-finding evaluation.

For example, say we have a bug benchmark `B` that defines a commit in its `project.yaml`:
1. We first use it as a coverage benchmark:
    * We build it with its `Dockerfile` and `build.sh` as usual but do not checkout to the commit in `project.yaml`.
3. We also use it as a bug benchmark:
    * We build it with its `Dockerfile` and `build.sh` as usual and then checkout to the commit in `project.yaml`.

This will not work for some bug benchmarks and that is fine: We only remove those and keep the ones that work.

Reply:
OK, would it be a good idea to have a coverage benchmark directory for `B` (e.g. `benchmarks/B/`) and another separate bug benchmark directory for the buggy commit of `B`(e.g. `benchmark/B_<commit/date/version>`)?

The downside is having to tolerate code duplication in their `Dockerfile` and `build.sh`.

Reply:
Based on our discussion today, I plan to proceed with the following steps:
1. Step 1: Treat all benchmarks are coverage benchmarks so that participants can test building their fuzzers with them.
    * Also upgrade bug benchmarks to the latest version (which is bug-free) so that participants can test them under the same environment.
2. Step 2: Temporarily remove the 1) commit hash and 2) `type: bug` label from their `benchmark.yaml` file.
3. Step 3: Find 5 benchmarks whose old buggy hash commit can work under the new environment.
4. Step 4: Add their corresponding 1) commit hash and 2) `type: bug` label back to their `benchmark.yaml` file.
5. Step 5: Find another 20 bug benchmarks.

Now:
1. [x] Step 1: We already did it in this PR.
2. [ ] Step 2: I will proceed if it looks good to you.
3. [ ] Step 3-5: These can be done later after we merge #1526 to the master branch.

Would this be a good plan to proceed? @jonathanmetzman 
Thanks!

Reply:
Sure!